pkg/controllers/namespace/namespace_controller.go

```diff
@@ -114,38 +128,157 @@ func getAccessMode(name string) v1alpha1.AccessMode {
 	}
 }
 
-func (r *NamespaceReconciler) createDefaultSubnetSet(ctx context.Context, ns string, defaultSubnetSize int) error {
+// GetNamespaceType determines the type of the namespace based on the VPCNetworkConfiguration
+func (r *NamespaceReconciler) getNamespaceType(ns *v1.Namespace, vnc *v1alpha1.VPCNetworkConfiguration) NameSpaceType {
+	anno := ns.Annotations
+	if len(anno) > 0 {
+		if ncName, exist := anno[types.AnnotationVPCNetworkConfig]; exist {
+			if ncName == "system" {
+				return SystemNs
+			}
+		}
+	}
+	label := ns.Labels
+	if len(label) > 0 {
+		if _, exist := label[SupervisorServiceIDLabel]; exist {
+			if value, exist := label["managedBy"]; exist && value == VsphereAppPlatformLabel {
+				return SVServiceNs
+			}
+		}
+	}
+	return NormalNs
+}
+
+func listSubnetSet(ctx context.Context, k8sClient client.Client, ns string, label client.MatchingLabels) (*v1alpha1.SubnetSet, error) {
+	var oldObj *v1alpha1.SubnetSet
+	list := &v1alpha1.SubnetSetList{}
+	if err := k8sClient.List(ctx, list, label, client.InNamespace(ns)); err != nil {
+		return nil, err
+	}
+	if len(list.Items) > 0 {
+		log.Info("Default SubnetSet already exists", "label", label, "Namespace", ns)
+		oldObj = &list.Items[0]
+	}
+	return oldObj, nil
+}
+
+func listDefaultSubnetSet(ctx context.Context, k8sClient client.Client, ns string, subnetSetType string) (*v1alpha1.SubnetSet, error) {
+	networkSubnetSetNameMap := map[string]string{
+		types.DefaultPodNetwork: types.LabelDefaultPodSubnetSet,
+		types.DefaultVMNetwork:  types.LabelDefaultVMSubnetSet,
+	}
+	label := client.MatchingLabels{
+		types.LabelDefaultNetwork: subnetSetType,
+	}
+	oldObj, err := listSubnetSet(ctx, k8sClient, ns, label)
+	if err != nil {
+		return nil, err
+	}
+	// check the old formatted default subnetset
+	if oldObj == nil {
+		newType, _ := networkSubnetSetNameMap[subnetSetType]
+		label = client.MatchingLabels{
+			types.LabelDefaultSubnetSet: newType,
+		}
+		return listSubnetSet(ctx, k8sClient, ns, label)
+	}
+	return oldObj, nil
+}
+
+func (r *NamespaceReconciler) getSystemNsDefaultSize() int {
+	defaultSubnetSize := util.MinSubnetSizeV90
+	if r.SubnetService.NSXClient.NSXCheckVersion(nsx.SubnetMinimalSize8) {
+		defaultSubnetSize = util.MinSubnetSizeV91
+	}
+	return defaultSubnetSize
+}
+
+func (r *NamespaceReconciler) getVMPodSubnet(sharedSubnet []v1alpha1.SharedSubnet) ([]string, []string) {
+	// Implementation to separate VM and Pod subnets from sharedSubnet
+	var vmSubnets []string
+	var podSubnets []string
+	for _, subnet := range sharedSubnet {
+		// Example condition to separate VM and Pod subnets
+		name := subnet.Name
+		if subnet.Name == "" {
+			items := strings.Split(subnet.Path, "/")
+			if len(items) > 0 {
+				name = items[len(items)-1]
+			}
+		}
+		if subnet.VMDefault == true {
+			vmSubnets = append(vmSubnets, name)
+		} else if subnet.PodDefault == true {
+			podSubnets = append(podSubnets, name)
+		}
+	}
+	return vmSubnets, podSubnets
```

Comment:
Maybe no need to return the subnet names, just return true or false for if default subnets exist for the two type?